File: app/src/main/java/de/garten/training/depotflow/data/db/converter/StopTypeConverter.java
package de.garten.training.depotflow.data.db.converter;

import org.greenrobot.greendao.converter.PropertyConverter;

import de.garten.training.depotflow.domain.StopType;

public class StopTypeConverter implements PropertyConverter<StopType, String> {

    @Override
    public StopType convertToEntityProperty(String databaseValue) {
        return StopType.fromDatabaseValue(databaseValue);
    }

    @Override
    public String convertToDatabaseValue(StopType entityProperty) {
        if (entityProperty == null) {
            return StopType.SERVICE.toDatabaseValue();
        }
        return entityProperty.toDatabaseValue();
    }
}

File: app/src/main/java/de/garten/training/depotflow/data/db/converter/StopTypeConverter.kt
package de.garten.training.depotflow.data.db.converter

import de.garten.training.depotflow.domain.StopType
import org.greenrobot.greendao.converter.PropertyConverter

class StopTypeConverter : PropertyConverter<StopType?, String?> {
    override fun convertToEntityProperty(databaseValue: String?): StopType? {
        return StopType.fromDatabaseValue(databaseValue)
    }

    override fun convertToDatabaseValue(entityProperty: StopType?): String? {
        if (entityProperty == null) {
            return StopType.SERVICE.toDatabaseValue()
        }
        return entityProperty.toDatabaseValue()
    }
}
